Running without dbus support

Fynedesk doesn't launch without dbus. Any possiblity to run it without dbus?

https://github.com/FyshOS/fynedesk/blob/master/wm/notifications.go

Here is the log

```
2025/10/12 14:45:12 Fyne error:  Unable to connect to session D-Bus
2025/10/12 14:45:12   Cause: exec: "dbus-launch": executable file not found in $PATH
2025/10/12 14:45:12   At: /home/hunter/go/pkg/mod/fyne.io/fyne/v2@v2.4.4/app/app_xdg.go:40
2025/10/12 14:45:12 Fyne error:  Unable to connect to session D-Bus
2025/10/12 14:45:12   Cause: exec: "dbus-launch": executable file not found in $PATH
2025/10/12 14:45:12   At: /home/hunter/go/pkg/mod/fyne.io/fyne/v2@v2.4.4/app/app_xdg.go:177
2025/10/12 14:45:12 Fyne error:  Error accessing to shared DBus connection
2025/10/12 14:45:12   Cause: exec: "dbus-launch": executable file not found in $PATH
2025/10/12 14:45:12   At: /home/hunter/soupbuild/BUILD/fynedesk-0.4.0/wm/dbus.go:45
2025/10/12 14:45:12 Fyne error:  Could not start DBus notifications server, using local only
2025/10/12 14:45:12   Cause: exec: "dbus-launch": executable file not found in $PATH
2025/10/12 14:45:12   At: /home/hunter/soupbuild/BUILD/fynedesk-0.4.0/wm/notifications.go:69
2025/10/12 14:45:12 Fyne error:  Could not launch xbacklight or brightnessctl
2025/10/12 14:45:12   Cause: exit status 1
2025/10/12 14:45:12   At: /home/hunter/soupbuild/BUILD/fynedesk-0.4.0/modules/status/brightness.go:200
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x70 pc=0x9f468a]

goroutine 1 [running, locked to thread]:
github.com/godbus/dbus/v5.(*Conn).export(0xca5600?, 0x1c29000?, {0xdc6d1b?, 0x41e865?}, {0xdd34f3?, 0xd36060?}, 0x1?)
	/home/hunter/go/pkg/mod/github.com/godbus/dbus/v5@v5.1.0/export.go:383 +0x2a
github.com/godbus/dbus/v5.(*Conn).ExportAll(0x0, {0xca5600?, 0x1c29000?}, {0xdc6d1b, 0x13}, {0xdd34f3, 0x1a})
	/home/hunter/go/pkg/mod/github.com/godbus/dbus/v5@v5.1.0/export.go:295 +0x65
fyshos.com/fynedesk/modules/systray.NewTray()
	/home/hunter/soupbuild/BUILD/fynedesk-0.4.0/modules/systray/main.go:76 +0x1fd
fyshos.com/fynedesk/internal/ui.(*desktop).Modules(0xc000346000)
	/home/hunter/soupbuild/BUILD/fynedesk-0.4.0/internal/ui/desk.go:247 +0x117
fyshos.com/fynedesk/internal/ui.(*background).loadModules(0xc00019ec80)
	/home/hunter/soupbuild/BUILD/fynedesk-0.4.0/internal/ui/background.go:29 +0x75
fyshos.com/fynedesk/internal/ui.(*background).CreateRenderer(0xcd6a80?)
	/home/hunter/soupbuild/BUILD/fynedesk-0.4.0/internal/ui/background.go:22 +0x17
fyne.io/fyne/v2/internal/cache.Renderer({0xed7788, 0xc00019ec80})
	/home/hunter/go/pkg/mod/fyne.io/fyne/v2@v2.4.4/internal/cache/widget.go:33 +0x147
fyne.io/fyne/v2/widget.(*BaseWidget).Resize(0xc00019ec80, {0x22fcd0?, 0xc0?})
	/home/hunter/go/pkg/mod/fyne.io/fyne/v2@v2.4.4/widget/widget.go:61 +0x11e
fyshos.com/fynedesk/internal/ui.(*desktop).Layout(0xc000346000, {0xc000201440?, 0x476d39?, 0x48?}, {0xd55460?, 0x0?})
	/home/hunter/soupbuild/BUILD/fynedesk-0.4.0/internal/ui/desk.go:92 +0x58
fyne.io/fyne/v2.(*Container).layout(...)
	/home/hunter/go/pkg/mod/fyne.io/fyne/v2@v2.4.4/container.go:195
fyne.io/fyne/v2.(*Container).Resize(0xc00022fd78?, {0x41e865?, 0x0?})
	/home/hunter/go/pkg/mod/fyne.io/fyne/v2@v2.4.4/container.go:168 +0x57
fyne.io/fyne/v2/internal/driver/glfw.(*glCanvas).SetContent(0xc0001a17c0, {0xed58a0, 0xc00019ecd0})
	/home/hunter/go/pkg/mod/fyne.io/fyne/v2@v2.4.4/internal/driver/glfw/canvas.go:152 +0x3a
fyne.io/fyne/v2/internal/driver/glfw.(*window).SetContent(0xc000362000, {0xed58a0, 0xc00019ecd0})
	/home/hunter/go/pkg/mod/fyne.io/fyne/v2@v2.4.4/internal/driver/glfw/window.go:260 +0xdf
fyshos.com/fynedesk/internal/ui.(*desktop).createRoot(0xc000346000, {0xc6d220?, 0x10?})
	/home/hunter/soupbuild/BUILD/fynedesk-0.4.0/internal/ui/desk.go:133 +0x4e
fyshos.com/fynedesk/internal/ui.(*desktop).setupRoot(0xc000346000)
	/home/hunter/soupbuild/BUILD/fynedesk-0.4.0/internal/ui/desk.go:140 +0x2f
fyshos.com/fynedesk/internal/ui.NewDesktop({0xedaf78?, 0xc0000fb280?}, {0xeda480?, 0xc000328000?}, {0xed4370?, 0xc000064348?}, {0xed4e08, 0xc0003d8000})
	/home/hunter/soupbuild/BUILD/fynedesk-0.4.0/internal/ui/desk.go:385 +0x135
main.setupDesktop({0xedaf78, 0xc0000fb280})
	/home/hunter/soupbuild/BUILD/fynedesk-0.4.0/cmd/fynedesk/main_unix.go:24 +0x196
main.main()
	/home/hunter/soupbuild/BUILD/fynedesk-0.4.0/cmd/fynedesk/main.go:19 +0x55
```

This is log for develop branch at @b4985efe

```
2025/10/12 15:08:12 Fyne error:  Error accessing to shared DBus connection
2025/10/12 15:08:12   Cause: exec: "dbus-launch": executable file not found in $PATH
2025/10/12 15:08:12   At: /home/hunter/Downloads/fynedesk-develop/wm/dbus.go:45
2025/10/12 15:08:12 Fyne error:  Could not start DBus notifications server, using local only
2025/10/12 15:08:12   Cause: exec: "dbus-launch": executable file not found in $PATH
2025/10/12 15:08:12   At: /home/hunter/Downloads/fynedesk-develop/wm/notifications.go:69
2025/10/12 15:08:12 Fyne error:  Could not launch xbacklight or brightnessctl
2025/10/12 15:08:12   Cause: exit status 1
2025/10/12 15:08:12   At: /home/hunter/Downloads/fynedesk-develop/modules/status/brightness.go:200
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x70 pc=0xb3856a]

goroutine 1 [running, locked to thread]:
github.com/godbus/dbus/v5.(*Conn).export(0xdcf9e0?, 0x1f22380?, {0xf165f8?, 0x420b05?}, {0xf23cbf?, 0xe73040?}, 0x1?)
	/home/hunter/go/pkg/mod/github.com/godbus/dbus/v5@v5.1.0/export.go:383 +0x2a
github.com/godbus/dbus/v5.(*Conn).ExportAll(0x0, {0xdcf9e0?, 0x1f22380?}, {0xf165f8, 0x13}, {0xf23cbf, 0x1a})
	/home/hunter/go/pkg/mod/github.com/godbus/dbus/v5@v5.1.0/export.go:295 +0x65
fyshos.com/fynedesk/modules/systray.NewTray()
	/home/hunter/Downloads/fynedesk-develop/modules/systray/main.go:76 +0x1fd
fyshos.com/fynedesk/internal/ui.(*desktop).Modules(0xc000464000)
	/home/hunter/Downloads/fynedesk-develop/internal/ui/desk.go:273 +0x117
fyshos.com/fynedesk/internal/ui.(*background).loadModules(0xc0004600c0)
	/home/hunter/Downloads/fynedesk-develop/internal/ui/background.go:29 +0x75
fyshos.com/fynedesk/internal/ui.(*background).CreateRenderer(0x10a8d20?)
	/home/hunter/Downloads/fynedesk-develop/internal/ui/background.go:22 +0x17
fyne.io/fyne/v2/internal/cache.Renderer({0x10a8d20, 0xc0004600c0})
	/home/hunter/go/pkg/mod/fyne.io/fyne/v2@v2.7.0-rc1/internal/cache/widget.go:20 +0x46
fyne.io/fyne/v2/widget.(*BaseWidget).Resize(0xc0001f7cf8?, {0x420b05?, 0x0?})
	/home/hunter/go/pkg/mod/fyne.io/fyne/v2@v2.7.0-rc1/widget/widget.go:51 +0x56
fyshos.com/fynedesk/internal/ui.(*desktop).Layout(0xc000464000, {0xc000460100?, 0x40?, 0xe85b40?}, {0x10a06c0?, 0x0?})
	/home/hunter/Downloads/fynedesk-develop/internal/ui/desk.go:96 +0x58
fyne.io/fyne/v2.(*Container).layout(...)
	/home/hunter/go/pkg/mod/fyne.io/fyne/v2@v2.7.0-rc1/container.go:185
fyne.io/fyne/v2.(*Container).Resize(0x420b05?, {0x40?, 0x0?})
	/home/hunter/go/pkg/mod/fyne.io/fyne/v2@v2.7.0-rc1/container.go:158 +0x57
fyne.io/fyne/v2/internal/driver/glfw.(*glCanvas).SetContent(0xc0005021a0, {0x10a65c0, 0xc000460140})
	/home/hunter/go/pkg/mod/fyne.io/fyne/v2@v2.7.0-rc1/internal/driver/glfw/canvas.go:140 +0x3a
fyne.io/fyne/v2/internal/driver/glfw.(*window).SetContent(0xc000502000, {0x10a65c0?, 0xc000460140?})
	/home/hunter/go/pkg/mod/fyne.io/fyne/v2@v2.7.0-rc1/internal/driver/glfw/window.go:230 +0x26
fyshos.com/fynedesk/internal/ui.(*desktop).createRoot(0xc000464000, {0xd91940?, 0x10?})
	/home/hunter/Downloads/fynedesk-develop/internal/ui/desk.go:146 +0x4e
fyshos.com/fynedesk/internal/ui.(*desktop).setupRoot(0xc000464000)
	/home/hunter/Downloads/fynedesk-develop/internal/ui/desk.go:153 +0x2f
fyshos.com/fynedesk/internal/ui.NewDesktop({0x10ad240?, 0xc0003529a0?}, {0x10ac950?, 0xc0004420d0?}, {0x10a5110?, 0xc000305dd0?}, {0x10a5b10, 0xc00045e000})
	/home/hunter/Downloads/fynedesk-develop/internal/ui/desk.go:394 +0x135
main.setupDesktop({0x10ad240, 0xc0003529a0})
	/home/hunter/Downloads/fynedesk-develop/cmd/fynedesk/main_unix.go:25 +0x196
main.main()
	/home/hunter/Downloads/fynedesk-develop/cmd/fynedesk/main.go:20 +0x58
```

Also wm/auth.go is using dbus.

```
2025/10/13 19:47:58 Fyne error:  Error accessing to shared DBus connection
2025/10/13 19:47:58   Cause: exec: "dbus-launch": executable file not found in $PATH
2025/10/13 19:47:58   At: /home/khalid/Downloads/fynedesk-fix-275/wm/dbus.go:45
2025/10/13 19:47:58 Fyne error:  Could not start DBus notifications server, using local only
2025/10/13 19:47:58   Cause: exec: "dbus-launch": executable file not found in $PATH
2025/10/13 19:47:58   At: /home/khalid/Downloads/fynedesk-fix-275/wm/notifications.go:69
2025/10/13 19:47:58 Fyne error:  Could not launch xbacklight or brightnessctl
2025/10/13 19:47:58   Cause: exit status 1
2025/10/13 19:47:58   At: /home/khalid/Downloads/fynedesk-fix-275/modules/status/brightness.go:200
2025/10/13 19:47:58 Fyne error:  Could not connect to DBus for system tray events
2025/10/13 19:47:58   Cause: exec: "dbus-launch": executable file not found in $PATH
2025/10/13 19:47:58   At: /home/khalid/Downloads/fynedesk-fix-275/modules/systray/main.go:75
2025/10/13 19:47:58 Fyne error:  Unable to start sound module
2025/10/13 19:47:58   Cause: dial unix /run/user/0/pulse/native: connect: no such file or directory
2025/10/13 19:47:58   At: /home/khalid/Downloads/fynedesk-fix-275/modules/status/sound.go:83
2025/10/13 19:47:58 Error running ifconfig tool exit status 127
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x70 pc=0xb3856a]

goroutine 59 [running]:
github.com/godbus/dbus/v5.(*Conn).export(0xe10a80?, 0xc0003ec170?, {0xf19319?, 0xb26760?}, {0xf39973?, 0xc0000bdec8?}, 0xb8?)
	/home/khalid/go/pkg/mod/github.com/godbus/dbus/v5@v5.1.0/export.go:383 +0x2a
github.com/godbus/dbus/v5.(*Conn).ExportAll(0x0, {0xe10a80?, 0xc0003ec170?}, {0xf19319, 0x14}, {0xf39973, 0x2e})
	/home/khalid/go/pkg/mod/github.com/godbus/dbus/v5@v5.1.0/export.go:295 +0x65
fyshos.com/fynedesk/wm.(*auth).register(0xc0003ec170)
	/home/khalid/Downloads/fynedesk-fix-275/wm/auth.go:38 +0x59
created by fyshos.com/fynedesk/wm.StartAuthAgent in goroutine 1
	/home/khalid/Downloads/fynedesk-fix-275/wm/auth.go:163 +0x89
```

--- a/modules/systray/main.go
+++ b/modules/systray/main.go
@@ -70,19 +70,23 @@ func NewTray() fynedesk.Module {
 	grid := container.New(collapsingGridWrap(fyne.NewSize(iconSize, iconSize)))
 	t := &tray{box: grid, nodes: make(map[dbus.Sender]*node)}
 
-	conn, _ := dbus.ConnectSessionBus()
+	conn, err := dbus.ConnectSessionBus()
+	if err != nil {
+		fyne.LogError("Could not connect to DBus for system tray events", err)
+		return t
+	}
 	t.conn = conn
 
-	err := conn.ExportAll(struct{}{}, hostPath, "org.kde.StatusNotifierHost")
+	err = conn.ExportAll(struct{}{}, hostPath, "org.kde.StatusNotifierHost")
 	if err != nil {
-		log.Println("Err", err)
+		fyne.LogError("Failed to export notifier host", err)
 		return t
 	}
 
 	// TODO this is create watcher (optional)
 	err = conn.ExportAll(t, path, "org.kde.StatusNotifierWatcher")
 	if err != nil {
-		log.Println("Err2", err)
+		fyne.LogError("Unable to register watcher", err)
 		return t
 	}
 

--- a/wm/auth.go
+++ b/wm/auth.go
@@ -34,8 +34,13 @@ type auth struct {
 }
 
 func (a *auth) register() {
-	conn2, _ := dbus.SystemBus()
-	err := conn2.ExportAll(a, "/AuthenticationAgent", "org.freedesktop.PolicyKit1.AuthenticationAgent")
+	conn2, err := dbus.SystemBus()
+	if err != nil {
+		fyne.LogError("Could not connect to DBus for authentication events", err)
+		return
+	}
+
+	err = conn2.ExportAll(a, "/AuthenticationAgent", "org.freedesktop.PolicyKit1.AuthenticationAgent")
 	if err != nil {
 		fyne.LogError("Could not start auth agent server", err)
 	}

